### PR TITLE
Add `flake8-bugbear` to the linter

### PR DESCRIPTION
**Pull request by Sjoerd1993**
`flake8-bugbear` also checks for certain patterns that are likely to cause bugs. The ones that did show up are addressed in this PR. 

See [link](https://github.com/PyCQA/flake8-bugbear) for more details.

### CODE REVIEW

**Comment by Sjoerd1993** [comment]
Couldn't get the test to fail locally, rerunning the test here gives a pass as well. Not sure why it failed that one run in that one commit.

File: hogben/optimise.py [hunk 1]
```diff
@@ -60,6 +61,9 @@ def optimise_angle_times(
         # Check that the measurement angle of the sample can be varied.
         assert isinstance(self.sample, VariableAngle)
 
+        # Set contrasts to empty list if not provided
+        contrasts = contrasts or []
```
Comment (jfkcooper) [inline hogben/optimise.py:65]: Is it not better to set the default in the init to `contrasts: Optional[list] = [],` this evaluates to `None` if required and removes this line

File: hogben/visualise.py [hunk 2]
```diff
@@ -51,6 +53,9 @@ def angle_choice(
     # Check that the angle can be varied for the sample.
     assert isinstance(sample, VariableAngle)
 
+    # Set contrasts to empty list if not provided
+    contrasts = contrasts or []
```
Comment (jfkcooper) [inline hogben/visualise.py:57]: Is it not better to set the default in the init to contrasts: Optional[list] = [], this evaluates to None if required and removes this line

File: hogben/visualise.py [hunk 3]
```diff
@@ -110,6 +115,9 @@ def angle_choice_with_time(
...
```
Comment (jfkcooper) [inline hogben/visualise.py:119]: Is it not better to set the default in the init to contrasts: Optional[list] = [], this evaluates to None if required and removes this line

**Review by jfkcooper — CHANGES_REQUESTED** [review]
three identical comments. I guess it is a linting thing, but it doesn't sit well with me to set a default to None, when we can set it to `[]` and not have any type uncertainty

**Comment by Sjoerd1993** [comment]
Sorry for being slightly late, but I've changed the default assignment to an else statement like `contrasts = [] if contrasts is None else contrasts`.

There's actually one issue that I saw [raised on StackOverflow](https://stackoverflow.com/questions/13710631/is-there-shorthand-for-returning-a-default-value-if-none-in-python) using the or operator, which I think is somewhat relevant:

`Note that this also returns "default" if x is any falsy value, including an empty list, 0, empty string, or even datetime.time(0) (midnight).`

As said before, `contrasts = contrasts or []` defaults to the first argument (so `contrasts = contrasts`) if `bool(contrasts) == True` and to the second argument (`contrasts = []`) if `bool(contrasts) == False`. This does lead to the (rare) scenario, that someone could parse `contrasts = 0`  as an argument, and that would pass perfectly fine without any errors, as `bool(0) == False`. So you really do lose a bit of your type safety.

This is not a problem using this slightly longer else statement. As that really enforces that argument actually _is_ the None object, and not just has an equal value or parses as the same boolean value. So it will only set `contrasts` to an empty list if `None` is actually given as argument (or when the argument is left blank obviously).
In practice it won't matter too much probably (it works perfectly fine, the only situation I could imagine this leading to a bug is someone setting contrasts to 0), but it does make the if/else construction a bit more robust anyway.

**Review by jfkcooper — APPROVED** [review]
This was something we should be careful of in future,, and something which I completely missed, so thanks for catching it and following up on it I guess your linting work has made it so we should catch these anyway

**Pull request merged**